pytest_metadata/plugin.py

```diff
@@ -68,7 +75,10 @@ def pytest_configure(config):
             "pluggy": pluggy.__version__,
         },
     }
-    config._metadata.update({k: v for k, v in config.getoption("metadata")})
+    metadata = {k: v for k, v in config.getoption("metadata")}
+    metadata_from_json = json_to_dict(config.getoption("metadata_from_json"))
```

Comment:
I'm putting this comment here and not in the function, because if you agree, there's no need for the function.

You shouldn't swallow user made errors. Which you're doing if you're catching the ValueError and returning the empty dict. At least not without issuing a warning to the user.

If there's a syntax error in the JSON, it will be very confusing to the user why the code runs fine - but his metadata has not been added.

I would just do:

´´´python
config._metadata.update(config.getoption("metadata_from_json"))
´´´

And just let an error bubble up naturally.

If you want to make a more human readable error, you can wrap the line in a try/except (no need for a function), but in this instance I would advise against it since there are so many types of syntax errors a user can make.

Reply:
Agreed. I had a custom exception at first, should have just left it 🤷.

Your code snippet here doesn't decode ```metadata_from_json``` string though. Is that a typo or am I missing something?

Reply:
Yeah, that's my bad. The `json.loads()` is missing. 😂

Reply:
Yeah, that's my bad. The `json.loads()` was missing. 😂